lib/services/credentials/credential.dao.js

```diff
@@ -124,7 +124,18 @@ dao.insertCredential = function (id, type, credentialObj) {
 };
 
 dao.getCredential = function (id, type) {
-  return db.hgetall(buildIdKey(type, id)).then(credential => {
+  const key = buildIdKey(type, id);
+  return db.type(key).then(type => {
+    if (type === 'hash') {
+      return db.hgetall(key);
+    } else if (type === 'set') {
+      throw Error(`Implementation error: ${key} is a set, therefore getAllCredentials() should be called instead of getCredential()`);
+    } else if (type === 'none') {
+      return null;
+    } else {
+      throw Error(`Corrupt data found at ${key}`);
```

Comment:
I believe this should be a new instance

```suggestion
      throw new Error(`Corrupt data found at ${key}`);
```